src/solvers/flattening/boolbv_width.cpp

```diff
@@ -194,10 +194,7 @@ const boolbv_widtht::entryt &boolbv_widtht::get_entry(const typet &type) const
     // no width
   }
   else if(type_id==ID_pointer)
-  {
-    entry.total_width=to_pointer_type(type).get_width();
-    DATA_INVARIANT(entry.total_width!=0, "pointer must have width");
-  }
+    entry.total_width=type_dynamic_cast<pointer_typet>(type).get_width();
```

Comment:
When is it considered ok to use a `type_dynamic_cast` and when should a `type_checked_cast` be used instead?

Reply:
`type_dynamic_cast` throws an exception if it cannot cast while `type_checked_cast` has a precondition. 
Actually I think the checked one would be better here since we don't catch the exception.